src/h3lib/lib/h3Index.c

```diff
@@ -1052,3 +1052,113 @@ H3Error H3_EXPORT(getPentagons)(int res, H3Index *out) {
  *         a Class II grid.
  */
 int isResolutionClassIII(int res) { return res % 2; }
+
+/**
+ * Returns the position of the cell within an ordered list of all children of
+ * the cell's parent at the specified resolution
+ */
+H3Error H3_EXPORT(cellToChildPos)(H3Index child, int parentRes, int64_t *out) {
+    int childRes = H3_GET_RESOLUTION(child);
+    // Get the parent at res. This will catch any resolution errors
+    H3Index parent;
+    H3Error parentError = H3_EXPORT(cellToParent(child, parentRes, &parent));
+    if (parentError) {
+        return parentError;
+    }
+    int parentIsPentagon = H3_EXPORT(isPentagon)(parent);
+
+    // Walk up the resolution digits, incrementing the index
+    *out = 0;
+    if (parentIsPentagon) {
+        // Pentagon logic. Pentagon parents skip the 1 digit, so the offsets are
+        // different from hexagons
+        for (int res = childRes; res > parentRes; res--) {
+            // It shouldn't be possible to get an error here, so we don't check
+            H3_EXPORT(cellToParent(child, res - 1, &parent));
+            parentIsPentagon = H3_EXPORT(isPentagon)(parent);
+            int rawDigit = H3_GET_INDEX_DIGIT(child, res);
+            int digit =
```

Comment:
What happens for invalid indexes into the deleted K subsequence of a pentagon, or indexes with INVALID_DIGIT (7)?

Reply:
Good questions:
- Invalid indexes into the K subsequence will be treated as `0` per the logic on the next line
- INVALID_DIGIT will result in an offset that's too high, and the resulting number might be greater than the number of max children

This _might_ be okay, on the principle of GIGO, but I could imagine attacks on systems _using_ H3 that would use poor output here to trigger random memory access. Possible failsafes:
- We could validate the digit and return an error
- We could clamp output to the max possible child index for the given parent

Validating the digits is probably worthwhile. If we do that we don't need to clamp output, but we could (after #720) add an assertion at the end of the function like
```
if (NEVER(out > cellToChildrenSize(parent, childRes))) {
  return E_FAILED;
}
```